### DISCUSSION TITLE
can you put the contract creation and the first input in a single transaction ?

### DISCUSSION THREAD

**Discussion by albert-kevin** [input-output-hk/marlowe-ts-sdk#154] [Q&A] [answered]
I wonder now if you can put runtimeLifecycle.contracts.createContract and runtimeLifecycle.contracts.applyInputs in a single transaction ?

**Comment by jhbertra** [comment, marked as answer]
Hi Kevin,

This is not currently possible using the APIs. We are working on adding support for arbitrary initial state, which will allow you to set up a contract with initial account balances, let-bindings, and choice values. This would add support for what you are asking about more directly. It also sidesteps some of the ambiguities involved with actually applying an input in the create transaction, such as how, if at all, to authenticate actions by parties, and what to do with the timeout information.
